Allow players to adjust the size of the ingame UI

pchote opened the pull request:
Finally fixes #10382.

~Depends on #17581.~

pchote committed:
Add Graphics.UIScale setting to modify UI size.

pchote commented:
Rebased.

pchote commented on OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs:
```diff
@@ -837,6 +853,78 @@ static void ShowBattlefieldCameraDropdown(DropDownButtonWidget dropdown, Graphic
 
+		static void RecalculateWidgetLayout(Widget w, bool insideScrollPanel = false)
+		{
+			// HACK: Recalculate the widget bounds to fit within the new effective window bounds
```
These hacks will go away once #16821 is ready.

Mailaender reviewed (dismissed):
Works as promised.

dragunoff commented:
The selection boxes and bars don't seem to render properly. Tested on Windows 7, changed UI scale in settings without changing the system DPI scale:
![selection-box-scale](https://user-images.githubusercontent.com/1355810/73476592-3a942280-439b-11ea-9e8f-9a3bde4626fb.png)

Or is that the expected result?

pchote commented:
This is the expected result - the weird triangular bits in the selection boxes is a long standing bug in the line renderer that has always been there with HiDPI rendering.

pchote committed:
Add UI Scale dropdown to the settings menu.

pchote commented:
Updated:
* Fixed the settings "reset" settings button putting the scale into an inconsistent state that broke ingame rendering
* Rebased on latest bleed.
